### PR TITLE
Fix gps circuit breaker logic in state machine.

### PR DESCRIPTION
**Pull request by jgoppert**
I wasn't able to fly with vision only in auto mission without this patch.

Also adds some strings to tell you why it is failsafe mode.

### CODE REVIEW

File: src/modules/commander/state_machine_helper.cpp [hunk 1]
```diff
@@ -80,6 +80,12 @@ using namespace DriverFramework;
 #define AVIONICS_WARN_VOLTAGE	4.9f
 #endif
 
+const char * reason_no_rc = "no rc";
```
Comment (julianoes) [inline src/modules/commander/state_machine_helper.cpp]: nitpick: should be `const char *reason_no_rc = "no rc";`

File: src/modules/commander/state_machine_helper.cpp [hunk 2]
```diff
@@ -651,6 +657,7 @@ bool set_nav_state(struct vehicle_status_s *status, struct commander_state_s *in
 		/* require RC for all manual modes */
 		if (rc_loss_enabled && (status->rc_signal_lost || status_flags->rc_signal_lost_cmd) && armed) {
 			status->failsafe = true;
+			PX4_WARN(reason_no_rc);
```
Comment (julianoes) [inline src/modules/commander/state_machine_helper.cpp]: These `PX4_WARN` are already much better, what do you think about having them report to mavlink as well?
However, that might be too verbose, not sure.

File: src/modules/commander/commander.cpp [hunk 3]
```diff
@@ -1341,6 +1341,11 @@ int commander_thread_main(int argc, char *argv[])
 	status_flags.circuit_breaker_engaged_gpsfailure_check = false;
 	get_circuit_breaker_params();
 
+	// initialize gps failure to false if circuit breaker enabled
+	if (status_flags.circuit_breaker_engaged_gpsfailure_check) {
+		status_flags.gps_failure = false;
+	}
```
Comment (julianoes) [inline src/modules/commander/commander.cpp]: I would prefer if this line was somehow together with:
https://github.com/PX4/Firmware/blob/d64c459b1b64cf35b501b43dc527ed430bad1f02/src/modules/commander/commander.cpp#L1318

What about making it an `else` case?

File: src/modules/commander/state_machine_helper.cpp (hunk 1 above)
Comment (jgoppert) [inline src/modules/commander/state_machine_helper.cpp]: Fixed.

File: src/modules/commander/commander.cpp (hunk 3 above)
Comment (jgoppert) [inline src/modules/commander/commander.cpp]: Fixed.

File: src/modules/commander/state_machine_helper.cpp (hunk 2 above)
Comment (julianoes) [inline src/modules/commander/state_machine_helper.cpp]: Cool!

**Comment by julianoes** [comment]
Reviewed, looks good to me. Let's test it and get it in.

**Pull request merged**